File: Lucidity/Tags.vb
Namespace Tags

  Public Class LucidityFolder
    Public Path As String = ""

    Public Sub New(ByVal Folder As String)
      Path = Folder
    End Sub
  End Class

  Public Class REMCyclesFolder
    Public Path As String = ""

    Public Sub New(ByVal Folder As String)
      Path = Folder
    End Sub
  End Class

  Public Class REMCycleFile
    Public Path As String = ""

    Public Sub New(ByVal Folder As String)
      Path = Folder
    End Sub
  End Class

  Public Class LucidItemFolder
    Public Path As String = ""
    Public Type As String = ""

    Public Sub New(ByVal Folder As String, ByVal strType As String)
      Path = Folder
      Type = strType
    End Sub
  End Class

  Public Class LucidItemFile
    Public Path As String = ""
    Public Type As String = ""

    Public Sub New(ByVal strPath As String, ByVal strType As String)
      Path = strPath
      Type = strType
    End Sub
  End Class


  Public Class DreamsFolder
    Public Path As String = ""

    Public Sub New(ByVal Folder As String)
      Path = Folder
    End Sub
  End Class

  Public Class YearFolder
    Public Path As String = ""
    Public Year As String = ""

    Public Sub New(ByVal Folder As String, ByVal strYear As String)
      Path = Folder
      Year = strYear
    End Sub
  End Class

  Public Class YearsFolder
    Public Path As String = ""
    Public Year As String = ""

    Public Sub New(ByVal Folder As String, ByVal strYear As String)
      Path = Folder
      Year = strYear
    End Sub
  End Class

  Public Class MonthFolder
    Public Path As String = ""
    Public Year As String = ""
    Public Month As String = ""

    Public Sub New(ByVal Folder As String, ByVal strYear As String, ByVal strMonth As String)
      Path = Folder
      Year = strYear
      Month = strMonth
    End Sub
  End Class

  Public Class DreamFile
    Public Path As String = ""

    Public Sub New(ByVal File As String)
      Path = File
    End Sub
  End Class
  

End Namespace

File: Lucidity/Tags.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
namespace lucidcode.LightenedDream.Lucidity.Tags
{

	public class LucidityFolder
	{

		public string Path = "";
		public LucidityFolder(string Folder)
		{
			Path = Folder;
		}
	}

	public class REMCyclesFolder
	{

		public string Path = "";
		public REMCyclesFolder(string Folder)
		{
			Path = Folder;
		}
	}

	public class REMCycleFile
	{

		public string Path = "";
		public REMCycleFile(string Folder)
		{
			Path = Folder;
		}
	}

	public class LucidItemFolder
	{
		public string Path = "";

		public string Type = "";
		public LucidItemFolder(string Folder, string strType)
		{
			Path = Folder;
			Type = strType;
		}
	}

	public class LucidItemFile
	{
		public string Path = "";

		public string Type = "";
		public LucidItemFile(string strPath, string strType)
		{
			Path = strPath;
			Type = strType;
		}
	}


	public class DreamsFolder
	{

		public string Path = "";
		public DreamsFolder(string Folder)
		{
			Path = Folder;
		}
	}

	public class YearFolder
	{
		public string Path = "";

		public string Year = "";
		public YearFolder(string Folder, string strYear)
		{
			Path = Folder;
			Year = strYear;
		}
	}

	public class YearsFolder
	{
		public string Path = "";

		public string Year = "";
		public YearsFolder(string Folder, string strYear)
		{
			Path = Folder;
			Year = strYear;
		}
	}

	public class MonthFolder
	{
		public string Path = "";
		public string Year = "";

		public string Month = "";
		public MonthFolder(string Folder, string strYear, string strMonth)
		{
			Path = Folder;
			Year = strYear;
			Month = strMonth;
		}
	}

	public class DreamFile
	{

		public string Path = "";
		public DreamFile(string File)
		{
			Path = File;
		}
	}


}
